onnxruntime/python/tools/transformers/benchmark.py

```diff
@@ -118,9 +118,10 @@ def run_onnxruntime(
         use_gpu
         and ("CUDAExecutionProvider" not in onnxruntime.get_available_providers())
         and ("ROCMExecutionProvider" not in onnxruntime.get_available_providers())
+        and ("DmlExecutionProvider" not in onnxruntime.get_available_providers())
```

Comment:
How about the following logic:
```
if (provider in ["CUDAExecutionProvider", "ROCMExecutionProvider", "DmlExecutionProvider"] and provider not in get_available_providers()):
```

Reply:
I don't think this works. Provider is generally a short string/identifier like "cuda", "rocm" or "dml", and it's also an optional parameter.